src/isl/mod.rs

```diff
@@ -30,66 +30,80 @@ pub mod isl_type_reference;
 #[cfg(test)]
 mod isl_tests {
     use crate::isl::isl_constraint::IslConstraint;
-    use crate::isl::isl_type::IslType;
+    use crate::isl::isl_type::{AnonymousIslType, IslType, NamedIslType};
     use crate::isl::isl_type_reference::IslTypeRef;
     use ion_rs::value::reader::element_reader;
     use ion_rs::value::reader::ElementReader;
     use ion_rs::IonType;
     use rstest::*;
-    use std::convert::TryInto;
 
-    // helper function to be used by isl tests
-    fn load(text: &str) -> IslType {
-        (&element_reader()
-            .read_one(text.as_bytes())
-            .expect("parsing failed unexpectedly"))
-            .try_into()
-            .unwrap()
+    // helper function to create NamedIslType for isl tests
+    fn load_named_type(text: &str) -> IslType {
+        IslType::NamedIslType(
+            NamedIslType::parse_from_owned_element(
+                &element_reader()
+                    .read_one(text.as_bytes())
+                    .expect("parsing failed unexpectedly"),
+            )
+            .unwrap(),
+        )
+    }
+
+    // helper function to create AnonymousIslType for isl tests
+    fn load_anonymous_type(text: &str) -> IslType {
+        IslType::AnonymousIslType(
+            AnonymousIslType::parse_from_owned_element(
+                &element_reader()
+                    .read_one(text.as_bytes())
+                    .expect("parsing failed unexpectedly"),
+            )
+            .unwrap(),
+        )
     }
 
     #[rstest(
     isl_type1,isl_type2,
     case::type_constraint_with_anonymous_type(
-    load(r#" // For a schema with single anonymous type
-            {type: int}
-        "#),
-    IslType::new(None, vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))])
+        load_anonymous_type(r#" // For a schema with single anonymous type
+                {type: int}
+            "#),
+        IslType::AnonymousIslType(AnonymousIslType::new(vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))]))
```

Comment:
Following other refactorings suggested above, this would look something like:
```rust
IslType::anonymous(
    vec![
        IslConstraint::type(
            IslTypeRef::core(IonType::Integer)
        )
    ]
)
```
(Spaced out more here for readability.)